### PR TITLE
159 create user input history component with clickable links

### LINKED ISSUE #159: Create User Input History Component with Clickable Links
**Related User Story**
#120 

**Related Tickets**
#148 #152 #160 

**Acceptance Criteria**
- Create a new component to display the user input history
- Integrate with the API service to fetch the user input history data from the backend
- Render each input history item as a clickable link

### PR DESCRIPTION
**Pull request by kinskiwu**
## Description
- Created UIs to display the user input list
- Implemented service to fetch user input list from API
- Implemented pagination with "Show more" and "Show less" functionality
- Implemented a skeleton UI to handle loading state
- Implemented simple styling
- Implemented basic axio config

image   image

Fixes #159 

## Type of change
- [X] ✨  New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- [X] Manual testing in dev mode

## Checklist:
- [X] I have performed a self-review of my own code.
- [X] I have commented on my code, particularly in hard-to-understand areas.
- [X] My changes generate no new warnings.
- [X] I have reviewed the [**Security Best Practices Document**](./docs/SECURITY_BEST_PRACTICES.md).
- [X] I have followed the [**Naming Conventions Guide**](./docs/NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [X] I have ensured that my pull request title is descriptive.

## Additional Information:
To be implemented in other tickets:
- Styling refactoring based on comments from this PR
- More robust setup for axios
- Integration with Zustand & React Query
- Integration with notification UI (global state needs to be set up before integration)

### CODE REVIEW

**Comment by kinskiwu** [comment]
Since this PR is getting quite large and further discussion on react query and zustand implementation (as @relandboyle mentioned at standup today) is pending, I will leave the remaining tasks for this ticket in separate PRs.

**Review by lomeintheory — APPROVED** [review]
I pulled down the branch and everything looks great! I assume the "..." icons are meant for a separate ticket to implement them according to the hamburger menu icons in our Figma design? I might've missed that during standups (Also, apparently it's called a kebab menu)

**Comment by kinskiwu** [comment]
@lomeintheory Thanks Andrew! Yep the `...` is for the hamburger/kebab menu ticket which I will work on next.

**Comment by kinskiwu** [comment]
Just realized the `...` should be vertical. Just updated the styling to be aligned with our [mockup](https://www.figma.com/design/9QKsvDQ9jivgA9XfS0oqPd/Design-System---Linkta?node-id=2453-7935&t=3iDVWT4nyrphoXjj-0)

File: client/hooks/useUserInputList.ts
Comment (kinskiwu) [inline client/hooks/useUserInputList.ts:1]: Will update this hook with react query in a future pr.

File: mocks/userInputList.ts
Comment (kinskiwu) [inline mocks/userInputList.ts:1]: Mock data for testing styling

File: server/controllers/userInputController.ts
Comment (kinskiwu) [inline server/controllers/userInputController.ts:1]: Fixed a bug & updated limit value

**Review by relandboyle — COMMENTED** [review]
@kinskiwu This looks really good, but... I've got some comments.

I think the "show more" button is meant to be a text button as per the Figma design. It should be closer to the bottom of the list, regardless of list length. Maybe it's even a special kind of list item, itself?

image

Also notice that "Recent" is a larger type and is aligned a bit farther left than the list items.

I think I see what you were going for with the up/down chevron icon, but from a UX standpoint it indicates a drop-down menu. The input history element may have some traits in common with a drop-down but it's also different in many ways.

The Typography elements are misaligned vertically with the icon buttons on the list items. This is due to bottom padding in the default text styles. The padding is useful in traditional text article layouts, but I prefer to disable it and allow direct control over text alignment. I added an override to our custom theme that handles this.

image

If you want to implement styles via the styled() method in the .tsx file, you can apply the global theme the way @lomeintheory did in the user input form.

The :hover style you've applied to the list items looks almost identical to the MUI button :hover style, which is very nice but I don't think it matches the Figma design.

image

**Comment by kinskiwu** [comment]
Thanks for the comment @relandboyle. I will incorporate your styling advice from this PR in subtask #160.

File: server/controllers/userInputController.ts
Comment (kinskiwu) [reply server/controllers/userInputController.ts:1]: @zhiyi15 Changes made here.

**Comment by kinskiwu** [comment]
Converted to draft to fix a rebase issue

**Comment by colinvandergraaf** [comment]
Aside from Reland's style comments, this looks good to go

**Review by colinvandergraaf — APPROVED** [review]

File: client/components/layout/UserInputHistory.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+const UserInputHistory = () => {
+  const { userInputList, loading, handleShowMore, handleShowLess, page } =
+    useUserInputList();
+  const [visibleItems, setVisibleItems] = useState(ITEMS_PER_PAGE);
+
+  useEffect(() => {
+    setVisibleItems(page * ITEMS_PER_PAGE);
+  }, [page]);
+
```
Comment (danielboisselle) [inline client/components/layout/UserInputHistory.tsx:47]: `const setVisibleItems = page + ITEMS_PER_PAGE;`

**Pull request merged**